### PR TITLE
Docs: rewriting the Column Summary topic and API ref

### PR DESCRIPTION
**Pull request by kirszenbaum**
This PR:
- Rewrites the [Column summary](http://localhost:8080/docs/next/column-summary/) topic
- Rewrites the Column summary [API ref](http://localhost:8080/docs/next/api/column-summary/)
- Adds missing options to the API ref

[skip changelog]

### CODE REVIEW

File: docs/next/guides/columns/column-summary.md [hunk 1]
```diff
@@ -14,334 +14,318 @@ tags:
...
+      destinationRow: 0,
+      destinationColumn: 1
     }
   ]
 });
 ```
-:::
 
-## Providing the settings as a function
+## Providing column summary configuration as a function
```
Comment (AMBudnik) [inline docs/next/guides/columns/column-summary.md:320]: The demo in this paragraph (in your code but also in the old version) replaces the first row of data with summaries. I think that we shouldn't do that.

**Review by AMBudnik — DISMISSED** [review]
Hey @kirszenbaum 

I have one suggestion (added to the code). Besides that, I approve all the changes 👍

File: docs/next/guides/columns/column-summary.md (hunk 1 above)
Comment (kirszenbaum) [reply docs/next/guides/columns/column-summary.md]: @AMBudnik - done: 99430eb. I also added a description and made a few minor changes

**Review by AMBudnik — APPROVED** [review]
Approved. Thank you for fixing the demo @kirszenbaum

**Pull request merged**